### PR TITLE
Move keyword primitive css dom

### LINKED ISSUE #85569: Generate primitive/keyword search result extra info into DOM instead of CSS
As discussed in #85568, once it's merged (it adds a test so it'll need to be updated), I'll remove this from the CSS and it will be generated into the DOM directly.

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Fixes #85569.

r? @jsha

### CODE REVIEW

File: src/librustdoc/html/static/search.js [hunk 1]
```diff
@@ -975,26 +975,32 @@ window.initSearch = function(rawSearchIndex) {
...
+                var type = itemTypes[item.ty];
+
                 length += 1;
 
+                var extra = "";
+                if (type === "primitive") {
+                    extra = " <i>(primitive type)</i>";
+                } else if (type === "keyword") {
+                    extra = " <i>(keyword)</i>";
+                }
```
Comment (jsha) [inline src/librustdoc/html/static/search.js:995]: Why declare a separate variable rather than add these directly to `name`?

Comment (GuillaumeGomez) [reply src/librustdoc/html/static/search.js:995]: Simpler to read in my opinion. So as your prefer. :)

Comment (jsha) [reply src/librustdoc/html/static/search.js:995]: If you prefer this way I'm fine with it. :-) Will r+ once conflicts are resolved.

Comment (GuillaumeGomez) [reply src/librustdoc/html/static/search.js:995]: Ok haha.

**Pull request merged**